book.md

```diff
@@ -495,17 +492,160 @@ probably include, or be followed by, a section on the various sanitizers
 available (ASan, UBSan, etc).}
 
 ## Code reuse attacks
-\missingcontent{Discuss ROP, JOP, COOP and mitigations (ASLR, CFI etc)}
+
+In the early days of memory vulnerability exploitation, attackers could simply
+place shellcode\index{shellcode} of their choice in executable memory and jump
+to it. As executable-space protection became mainstream, attackers started to
+reuse code already present in an application's binary and linked libraries
+instead. A variety of different techniques to this effect came to light.
+
+The simplest of these techniques is return-to-libc [@Solar1997]. Instead of
+returning to shellcode that the attacker has injected, the return address is
+modified to return into a library function, such as `system` or `exec`. This
+technique is simpler to use when arguments are also passed on the stack and
+can therefore be controlled with the same stack buffer overflow that is used
+to modify the address.
+
+Return-to-libc attacks restrict an attacker to whole library functions. While
+this can lead to powerful attacks, it has also been demonstrated that it is
+possible to achieve arbitrary computation by combining a number of short
+instruction sequences known as **gadgets**\index{gadget}. In return-oriented
+programming (ROP)\index{return-oriented programming (ROP)} [@Shacham2007], In
+ROP, each gadget performs a simple operation, for example setting a register,
+then pops a return address from the stack and returns to it. The attacker
+constructs a fake call stack (often called a ROP chain) which ensures a number
+of gadgets are executed one after another, in order to perform a more complex
+operation.
+
+This will hopefully become more clear with an example: a ROP chain for
+AArch64 that starts a shell, by calling `execve` with `"/bin/sh"` as an
```

Comment:
Maybe be more explicit and say "aarch64-linux" rather than just "AArch64", as I guess the example might not work on other OSes such as Windows?